Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -17,6 +17,118 @@ import NIO
 let crlf: StaticString = "\r\n"
 let headerSeparator: StaticString = ": "
 
+/// An `IteratorProtocol` that can iterate through comma separated list of values for a certain
+/// header
+///
+///  **Example:**
+///
+/// Suppose you have this headers:
+///
+///      Connection: keep-alive, x-server
+///      Content-Type: text/html
+///      Connection: other
+///
+/// You can iterate using this struct on those headers, for values of `Connection`, to get
+/// `keep-alive`, then `x-server`, then `other`
+///
+public struct HTTPListHeaderIterator: IteratorProtocol {
+    
+    public typealias Element = ByteBuffer
+    
+    fileprivate static let comma = UInt8(",".utf8CString[0])
+    fileprivate static let whiteSpaces = [" ", "\t"]
+                                             .map( {UInt8($0.utf8CString[0])} )
+    
+    private var isNowOnAField = false
+    private var currentHeaderIndex: Int = -1
+    private var singleValueParser: ByteBufferSliceSplitIterator
+    private var byteBuffer: ByteBuffer
+    private let headerName: ContiguousArray<UInt8>
+    private let headers: [HTTPHeader]
+    
+    /// Returns next index in headers
+    ///
+    /// - Parameter current: The index to begin iteration at
+    /// - Returns: The next index of the header in header array, or `nil` if not found
+    public func nextHeaderIndex(ifCurrentIs current: Int) -> Int? {
+        var ret: Int? = nil
+        for (idx, currentHeader) in headers.enumerated().dropFirst(current + 1) {
+            if byteBuffer.getSlice(at: currentHeader.name.start,
+                                   length: currentHeader.name.length)?
+                .compareReadableBytes(to: headerName) ?? false {
+                ret = idx
+                break
+            }
+        }
+        return ret
+    }
+
+    mutating public func next() -> ByteBuffer? {
+        // Still not reading a specific field
+        if !isNowOnAField {
+            // Try to get next match
+            guard let index = nextHeaderIndex(ifCurrentIs: currentHeaderIndex) else { return nil }
+            // If succeeded, prepare for parsing it
+            currentHeaderIndex = index
+            isNowOnAField = true
+            singleValueParser = ByteBufferSliceSplitIterator(
+                byteBuffer: byteBuffer,
+                separator: HTTPListHeaderIterator.comma,
+                start: headers[currentHeaderIndex].value.start,
+                length: headers[currentHeaderIndex].value.length)
+        }
```

Comment:
Rather than use a `ByteBufferSliceSplitIterator`, we can split the `ByteBufferView` directly. Because the `ByteBufferView` doesn't copy the data doing the split eagerly is not too expensive, and greatly simplifies the code.